corehq/apps/cloudcare/static/cloudcare/js/README.md

```diff
@@ -48,7 +54,33 @@ These are HQ apps. Most of the logic around apps has to do with displaying the h
+
+* A CommCareHQ user goes to the module settings page in app builder and turns on the feature, selecting the registration form they want to be accessible from the case list.
+   * This adds a new attribute to their `Application` document - specifically, it populates `case_list_form` on a `Module`.
+* When the user makes a new build of their app, the app building code reads the `Application` doc and writes out all of the application files, including the `suite.xml`.
+   * The module's case list configuration is transformed into a [detail](https://github.com/dimagi/commcare-core/wiki/Suite20#detail) element, which includes an [action](https://github.com/dimagi/commcare-core/wiki/Suite20#action) element that represents the case list form.
+* When a Web Apps user clicks the menu's name to access the case list, web apps sends a `navigate_menu` request to formplayer that includes a set of `selections` (see [navigation and replaying of sessions](https://github.com/dimagi/commcare-hq/blob/master/docs/formplayer.rst#navigation-and-replaying-of-sessions)).
+   * The formplayer response tells web apps what kind of sceen to display:
+      * The `type` is `entities` which tells web apps to display a case list UI
+      * The `entities` list contains the cases and their properties
+      * The `actions` list includes an action for the case list registration form, which tells web apps to display a button at the bottom of the case list with the given label, that when clicked will add the string `action 0` to the `selections` list and then send formplayer another navigation request, which will cause formplayer to send back a form response for the registration form, which web apps will then display for the user.
+
+Note how generic the concepts web apps deals with are: "entities" can be cases, fixture rows, ledger values, etc. Web apps doesn't know what cases are, and it doesn't know the difference between an action that triggers a case list registration form and an action that triggers a case search.
+
+Development for most new web apps features maps to the steps above, requiring some or all of the following:
+Logic | Repository | Language
+----- | ---------- | --------
+App manager UI where the the feature is enabled & configured | commcare-hq | Python / HTML / JavaScript
+Updated logic to make a new app build, typically changes to suite generation | commcare-hq | Python
+New model for the configuration | commcare-core | Java
+Formplayer processing to add the new feature to the appropriate response | formplayer | Java
+Web apps UI for the feature | commcare-hq | JavaScript / HTML
+CommCare Mobile UI for the new feature | commcare-android | Java
```

Comment:
:grimacing:

Reply:
yeah...I think web apps takes time to get used to less because it's heavy on javascript (although I'm biased) and more because there are so many different subsystems involved.